docs/cpp/move-constructors-and-move-assignment-operators-cpp.md

```diff
@@ -125,14 +125,20 @@ The following procedures describe how to write a move constructor and a move ass
     }
     ```
 
-1. In the move assignment operator, add a conditional statement that performs no operation if you try to assign the object to itself.
+1. In the move assignment operator, we need to keep the self-assignment safe, and the simple way is to add a judgment directly.
 
     ```cpp
     if (this != &other)
     {
     }
     ```
 
+    There are many other ways, too, such as `copy-and-swap`.
+
+    ```cpp
+    shared_ptr(_Right).swap(*this);
```

Comment:
It's a bit weird to mention `shared_ptr` here. And `shared_ptr(_Right).swap(*this)` is performed in copy assignment, not in move assigment.

In this `move-constructors-and-move-assignment-operators-cpp.md`, I think we should say `MemoryBlock(std::move(other)).swap(*this)`. BUT we also need to add a `swap` function together.